src/compositions/useSavedFlowRunsSearches.ts

```diff
@@ -0,0 +1,61 @@
+import { showToast } from '@prefecthq/prefect-design'
+import { useSubscription } from '@prefecthq/vue-compositions'
+import { computed, ComputedRef, watch } from 'vue'
+import { useCustomDefaultFlowRunsFilter } from '@/compositions/useCustomDefaultFlowRunsFilter'
+import { useWorkspaceApi } from '@/compositions/useWorkspaceApi'
+import { FlowRunsFilter, SavedSearch, SavedSearchCreate } from '@/models'
+import { mapper } from '@/services'
+import { builtinSavedSearches, customSavedSearch, isSameFilter, systemDefaultSavedSearch } from '@/utilities/savedFilters'
+
+export type SavedFlowRunsSearch = SavedSearch & { isDefault: boolean }
+
+type UseSavedFlowRunsSearches = {
+  savedFlowRunsSearches: ComputedRef<SavedFlowRunsSearch[]>,
+  createSavedFlowRunsSearch: (search: SavedSearchCreate) => Promise<SavedSearch>,
+  deleteSavedFlowRunsSearch: (savedSearchId: string) => Promise<void>,
+  default: ComputedRef<FlowRunsFilter>,
+  hasCustomDefault: ComputedRef<boolean>,
+}
+
+export function useSavedFlowRunsSearches(): UseSavedFlowRunsSearches {
+  const api = useWorkspaceApi()
+  const savedSearchesSubscription = useSubscription(api.savedSearches.getSavedSearches)
+  const savedSearches = computed(() => builtinSavedSearches.concat(savedSearchesSubscription.response ?? []))
+
+  const { value: myCustomDefaultSavedSearchFilter, remove: removeDefault } = useCustomDefaultFlowRunsFilter()
+  const hasCustomDefault = computed(() => myCustomDefaultSavedSearchFilter.value !== null)
+  const myDefaultSavedFilter = computed(() => myCustomDefaultSavedSearchFilter.value ?? systemDefaultSavedSearch.filters)
+  const defaultFlowRunsFilter = computed(() => mapper.map('SavedSearchFilter', myDefaultSavedFilter.value, 'FlowRunsFilter'))
+
+  const matchedCustomDefaultSavedSearch = computed(() => savedSearches.value.find(search => search.name !== customSavedSearch.name && isSameFilter(search.filters, myDefaultSavedFilter.value)))
+  const savedFlowRunsSearches = computed(() => savedSearches.value.map(search => ({ ...search, isDefault: matchedCustomDefaultSavedSearch.value?.name === search.name })))
+
+  // handle if customDefault is not in savedSearches i.e. someone else deleted it
+  watch([matchedCustomDefaultSavedSearch, savedSearchesSubscription], ([found, subscription]) => {
+    if (!found && subscription.executed) {
+      showToast('Your custom default flow runs filter was deleted by another user. Resave it or select a new default.')
+      removeDefault()
```

Comment:
Now that I see this toast implementation I think we should remove this. Even if the saved version of a users custom default is deleted. We can keep loading their custom default from local storage indefinitely. 

The only thing we'd need to account for is if we load a custom default (that doesn't match a saved search) and then the user changes the filters. We'll end up with two "Custom" filters. Which honestly is probably fine. But maybe we have a special name for custom default filters that don't match a saved search.

Reply:
👍 

<img width="633" alt="image" src="https://github.com/PrefectHQ/prefect-ui-library/assets/22418768/decb6f00-9f1a-4a19-af92-007477fcc9d8">

Reply:
Nice, I like that. "Unsaved (default)" makes sense to me 👍